EIPS/eip-7775.md

```diff
@@ -0,0 +1,119 @@
+5. If the amount to be burned is 0, the execution `MUST NOT` revert.
+6. Subtracts the amount from the current address's native ether balance.
+
+When `BURN` is used in the context of a DELEGATECALL or CALLCODE, the contract whose balance is to be manipulated is the contract that issued the DELEGATECALL or CALLCODE instruction.
+
+When `BURN` is used in the context of a STATICCALL the call `MUST` revert.
+
+### Gas Cost
+
+The base gas cost for the `BURN` opcode is 100 gas. The dynamic gas cost is determined as follows:
+
+1. If the value to be burned is 0, the dynamic gas cost is 0.
+2. Otherwise, the dynamic gas cost is 100.
+
+The total gas cost for the `BURN` opcode is the sum of the base gas cost and the dynamic gas cost.
+
+### Pseudocode
+
+Example pseudocode for the `BURN` opcode:
+
+```python
+def op_burn(pc, interpreter, scope):
+    # Consume the base gas cost
+    interpreter.consume_gas(100)
+
+    # Pop the value to be burned from the stack
+    value_to_burn = scope.stack.pop()
+
+    # If the value to be burned is 0, do not revert
+    if value_to_burn == 0:
+        return None
+
+    # Retrieve the current address from the EVM execution context
+    current_address = scope.contract.address()
+
+    # Check the balance of the current address
+    balance = interpreter.evm.state_db.get_balance(current_address)
+
+    # If the value to be burned is greater than the balance, revert
+    if value_to_burn > balance:
+        return "ErrInsufficientBalance"
+
+    # Subtract the value from the current address's balance
+    interpreter.evm.state_db.sub_balance(current_address, value_to_burn)
+   
+    # Consume the dynamic gas cost
+    interpreter.consume_gas(100)
+    return None
+```
+
+## Rationale
+
+The introduction of the `BURN` opcode helps clean up a piece of weird semantics in the Ethereum. Historically, burning native ether involved sending them to an address from which they could not be recovered, such as the zero address or a contract with no withdrawal functionality. This method is not only inefficient but also confusing for indexers and other tools that track token movements. By providing a dedicated `BURN` opcode, we eliminate this ambiguity and ensure that the act of burning tokens is explicit and standardized.
+
+Potential Pros:
+
+- Provides a clear and standardized method for burning native ether within smart contracts.
+- Allows for better accounting practices of native token within smart contracts.
+- Reduces the possibility of a smart contract exploit caused by native token that was marked as "burned" being unintetionally recovered.
+
+Potential Cons:
+
+- Does not help remove unrecoverable ether sitting in existing contracts.
+- New code in the clients
+- New concept needed to be added to the yellow paper.
+
+## Backwards Compatibility
+
+This EIP introduces a new opcode and thus must be activated via a scheduled hardfork.
+
+## Test Cases
+
+- Coming soon
```

Comment:
put a html tag `<!-- TODO -->` which is tracked by the bots to make sure its addressed before going to review etc

Reply:
👍